tests/test_utils.py

```diff
@@ -730,3 +713,32 @@ def test_authorize_pygsheets_raises_after_failing_both(self, mocker, caplog):
             mocker.call(service_file='credentials'),
             mocker.call(custom_credentials='credentials')
         ]
+
+
+class TestEmptyStringsAsNulls:
+
+    def test_converts_empty_strings_to_null(self):
+        FeatureSet = namedtuple('FeatureSet', ['features'])
+        Feature = namedtuple('Feature', ['attributes'])
+        feature_set = FeatureSet([Feature({
+            'a': 'foo',
+            'b': 'baz',
+        }), Feature({
+            'a': '',
+            'b': '',
+        })])
+        fields = [{
+            'name': 'a',
+            'type': 'esriFieldTypeInteger',
+            'nullable': False,
+        }, {
+            'name': 'b',
+            'type': 'esriFieldTypeInteger',
+            'nullable': True,
+        }]
+
+        fixed_feature_set = palletjack.utils.fix_numeric_empty_strings(feature_set, fields)
+        fixed_feature = fixed_feature_set.features[1]
+
+        assert fixed_feature.attributes['a'] == ''  #: perhaps these should be set to 0 or -999?
```

Comment:
```suggestion
        assert fixed_feature.attributes['a'] == ''
```
v3 has a separate check earlier in the pipeline for Null/None/NaNs in dataframe columns that map to fields that aren't nullable and throws an exception there. I think it's a data quality issue that's best handled by the user/client program, not a fix around how the feature service expects the data to look.